src/patches/key.cc

```diff
@@ -0,0 +1,278 @@
+    {">", KeyCode::Greater},
+    {"|", KeyCode::Pipe},
+    {"~", KeyCode::Tilde},
+    {"$", KeyCode::Dollar},
+    {"0", KeyCode::Alpha0},
+    {"1", KeyCode::Alpha1},
+    {"2", KeyCode::Alpha2},
+    {"3", KeyCode::Alpha3},
+    {"4", KeyCode::Alpha4},
+    {"5", KeyCode::Alpha5},
+    {"6", KeyCode::Alpha6},
+    {"7", KeyCode::Alpha7},
+    {"8", KeyCode::Alpha8},
+    {"9", KeyCode::Alpha9},
+    {"A", KeyCode::A},
+    {"B", KeyCode::B},
+    {"C", KeyCode::C},
+    {"D", KeyCode::D},
+    {"E", KeyCode::E},
+    {"F", KeyCode::F},
+    {"F1", KeyCode::F1},
+    {"F10", KeyCode::F10},
+    {"F11", KeyCode::F11},
+    {"F12", KeyCode::F12},
+    {"F13", KeyCode::F13},
+    {"F14", KeyCode::F14},
+    {"F15", KeyCode::F15},
+    {"F2", KeyCode::F2},
+    {"F3", KeyCode::F3},
+    {"F4", KeyCode::F4},
+    {"F5", KeyCode::F5},
+    {"F6", KeyCode::F6},
+    {"F7", KeyCode::F7},
+    {"F8", KeyCode::F8},
+    {"F9", KeyCode::F9},
+    {"G", KeyCode::G},
+    {"H", KeyCode::H},
+    {"I", KeyCode::I},
+    {"J", KeyCode::J},
+    {"K", KeyCode::K},
+    {"L", KeyCode::L},
+    {"M", KeyCode::M},
+    {"N", KeyCode::N},
+    {"O", KeyCode::O},
+    {"P", KeyCode::P},
+    {"Q", KeyCode::Q},
+    {"R", KeyCode::R},
+    {"S", KeyCode::S},
+    {"T", KeyCode::T},
+    {"U", KeyCode::U},
+    {"V", KeyCode::V},
+    {"W", KeyCode::W},
+    {"X", KeyCode::X},
+    {"Y", KeyCode::Y},
+    {"Z", KeyCode::Z},
+    {"NUMLOCK", KeyCode::Numlock},
+    {"KEY0", KeyCode::Keypad0},
+    {"KEY1", KeyCode::Keypad1},
+    {"KEY2", KeyCode::Keypad2},
+    {"KEY3", KeyCode::Keypad3},
+    {"KEY4", KeyCode::Keypad4},
+    {"KEY5", KeyCode::Keypad5},
+    {"KEY6", KeyCode::Keypad6},
+    {"KEY7", KeyCode::Keypad7},
+    {"KEY8", KeyCode::Keypad8},
+    {"KEY9", KeyCode::Keypad9},
+    {"KEYDIVIDE", KeyCode::KeypadDivide},
+    {"KEYENTER", KeyCode::KeypadEnter},
+    {"KEYEQUAL", KeyCode::KeypadEquals},
+    {"KEYMINUS", KeyCode::KeypadMinus},
+    {"KEYMULTI", KeyCode::KeypadMultiply},
+    {"KEYPERIOD", KeyCode::KeypadPeriod},
+    {"KEYPLUS", KeyCode::KeypadPlus},
+};
+
+const std::vector<KeyCode> Key::modifiers = {
+    KeyCode::LeftAlt,
+    KeyCode::LeftApple,
+    KeyCode::LeftCommand,
+    KeyCode::LeftControl,
+    KeyCode::AltGr,
+    KeyCode::LeftShift,
+    KeyCode::LeftWindows,
+    KeyCode::RightAlt,
+    KeyCode::RightApple,
+    KeyCode::RightCommand,
+    KeyCode::RightControl,
+    KeyCode::RightShift,
+    KeyCode::RightWindows,
+};
+
+Key::Key()
+{
+  // Map();
+}
+
+KeyCode Key::Parse(std::string_view key)
+{
+  auto wantedKey = absl::AsciiStrToUpper(key);
+  for (const auto& [value, keycode] : mappedKeys) {
+    if (key == value) {
+      return (KeyCode)keycode;
+    }
+  }
+
+  return KeyCode::None;
+}
+
+bool Key::IsModifier(KeyCode key)
+{
+  return (std::find(Key::modifiers.begin(), Key::modifiers.end(), key) != Key::modifiers.end());
+}
+
+bool Key::IsModified()
+{
+  for (const auto modifier: Key::modifiers) {
```

Comment:
I get the feeling that we should probably cache modified on a per-frame basis. Doing repeated calls into the engine for this is most likely a performance cliff.